site/src/components/entityTypes/SchemaEditor/SchemaEditor.tsx

```diff
@@ -0,0 +1,334 @@
+
+      if (!baseUrl && ($ref.startsWith("#") || $ref.startsWith("/"))) {
+        throw new Error(
+          `Cannot resolve relative $ref '${$ref} as the schema has no absolute $id to resolve it against.`,
+        );
+      }
+
+      let schemaLinkPath = "";
+      /**
+       * @todo catch links to schemas served from outside blockprotocol.org, and instead of opening their off-site pages,
+       *    fetch them and load them into our viewer. Will need to update relative approaches too.
+       */
+      if ($ref.startsWith("#")) {
+        /**
+         * This is a relative link to a sub-schema of this same schema
+         * @see https://json-schema.org/understanding-json-schema/structuring.html#json-pointer
+         */
+        schemaLinkPath = schema$id + $ref;
+      } else if ($ref.startsWith("/")) {
+        /**
+         * This is a relative link to another schema to be resolved against the base URL of this schema.
+         * @see https://json-schema.org/understanding-json-schema/structuring.html#ref
+         */
+        schemaLinkPath = baseUrl + $ref;
+      } else if ($ref.startsWith("http")) {
+        schemaLinkPath = $ref;
+      } else {
+        /**
+         * This could be a property name for an object defined in the tree of the schema or a sub-schema within it.
+         * Really these should instead be defined under $defs and referenced as such, but they might exist.
+         */
+        schemaLinkPath = `${
+          schema$id + (subSchemaReference || "#")
+        }/properties/${$ref}`;
+      }
+
+      return (
+        <Link href={schemaLinkPath}>
+          <a>
+            <strong>{$ref.replace(/#\/\$defs\//g, "")}</strong>
+          </a>
+        </Link>
+      );
+    },
+    [schema$id, subSchemaReference],
+  );
+
+  /**
+   * @todo deal with $anchors https://json-schema.org/understanding-json-schema/structuring.html#anchor
+   */
+  const selectedSchema: JsonSchema = pathToSubSchema
+    ? get(workingSchemaDraft, pathToSubSchema)
+    : workingSchemaDraft;
+
+  const schemaOptions = useMemo(
+    () => ({
+      availableEntityTypes: availableEntityTypes ?? [],
+      generateAbsoluteSchemaLink,
+      selectedSchema,
+      subSchemas,
+    }),
+    [
+      availableEntityTypes,
+      generateAbsoluteSchemaLink,
+      selectedSchema,
+      subSchemas,
+    ],
+  );
+
+  const { description } = selectedSchema;
+
+  const readonly = !updateEntityTypes || !entityTypeId;
+
+  // const addSubSchema = (newSubSchemaName: string) =>
+  //   dispatchSchemaUpdate({
+  //     type: "addSubSchema",
+  //     payload: { newSubSchemaName },
+  //   });
+  //
+  // const [newSubSchemaName, setNewSubSchemaName] = useState("");
+
+  // const onNewSubSchemaFormSubmit = (event: FormEvent<HTMLFormElement>) => {
+  //   event.preventDefault();
+  //
+  //   if (!newSubSchemaName.trim()) {
+  //     return;
+  //   }
+  //
+  //   addSubSchema(newSubSchemaName);
+  //   setNewSubSchemaName("");
+  // };
+
```

Comment:
```suggestion
```

(alternatively we can add a comment saying when to uncomment)

Reply:
I've added a comment - I did start removing this code yesterday but it was going to involve digging through more components, and it might just have to be all restored soon. So I've added comments/`@todos` for now